fuel-core/src/model/fuel_block.rs

```diff
@@ -67,31 +69,73 @@ impl BlockHeight {
     }
 }
 
+#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
+pub struct TransactionCommitment {
+    pub sum: Word,
+    pub root: Bytes32,
+}
+
 #[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct FuelBlock {
+pub struct FuelBlockHeaders {
     pub fuel_height: BlockHeight,
-    pub transactions: Vec<Bytes32>,
     pub time: DateTime<Utc>,
     pub producer: Address,
+    // TODO: integrate with fuel-merkle
+    pub transactions_commitment: TransactionCommitment,
 }
 
-impl Default for FuelBlock {
+impl FuelBlockHeaders {
+    pub fn id(&self, transaction_ids: &[Bytes32]) -> Bytes32 {
+        let mut hasher = Hasher::from_iter(transaction_ids);
+        hasher.input(&self.fuel_height.to_bytes()[..]);
+        hasher.input(self.time.timestamp_millis().to_be_bytes());
+        hasher.input(self.producer.as_ref());
+        hasher.input(self.transactions_commitment.sum.to_be_bytes());
+        hasher.input(self.transactions_commitment.root.as_ref());
+        hasher.digest()
```

Comment:
I think the block id should always be the same once the block is produced. And actually, it shouldn't take the tx ids anymore since the TransactionCommitment MST should already include all the tx data.